Policies: pre-annotated services may be removed when all-allowing policy in place and skupper initialized

# How to reproduce

On a cluster with the Policy CRD and an all-allowing Policy CR installed, run the test `TestAnnotatedResources/services-pre-annotated` (at `test/integration/acceptance/annotated_resource_test.go`)

Job aid:

    go test ./test/integration/acceptance/... -tags integration -timeout 720m -v -count 1 -p 1 -run TestAnnotatedResources/services-pre-annotated

The test will (most probably) end in timeout, with an error like the one below:

```
2022/05/13 02:27:22 Trying to retrieve ConsoleData...
2022/05/13 02:27:24 ConsoleData do not match yet
2022/05/13 02:27:24 Number of sites  [expected: 2 - found: 2]
2022/05/13 02:27:24 Exposed services [expected: 10 - found: 8]
2022/05/13 02:27:27 Trying to retrieve ConsoleData...
2022/05/13 02:27:30 ConsoleData do not match yet
2022/05/13 02:27:30 Number of sites  [expected: 2 - found: 2]
2022/05/13 02:27:30 Exposed services [expected: 10 - found: 8]
(...)
--- FAIL: TestAnnotatedResources (700.28s)
    --- FAIL: TestAnnotatedResources/services-pre-annotated (602.07s)
        annotated_resource_test.go:179: assertion failed: error is not nil: context deadline exceeded: timed out waiting for expected services/protocol list to match
```

The actual number of found services will vary from test to test (it will sometimes even get to 10, so you may need to retest). 

# Additional information

- Looking at the output, the number of found services may have gone higher than the final count
- Running `get events`, there are `PolicyController` events happening later than the service definition updates and service creations:
  - `Skupper policy has changed: background-policy`
  - `[validateServiceStateChanged] service is now allowed nginx-2-ds-web`
- There are `ServiceControllerDeleteEvent` events reporting the service removals:
  -  `Deleting service nginx-2-ds-web`
  - `No service binding found for nginx-2-ds-web`
- In at least one of the runs, there were events logging a failure to update config map:
  - `Failed to update skupper-services config map: Operation cannot be fulfilled on configmaps "skupper-services": the object has been modified; please apply your changes to the latest version and try again`

--- a/cmd/service-controller/policy_controller.go
+++ b/cmd/service-controller/policy_controller.go
@@ -384,7 +384,7 @@ func (c *PolicyController) validateServiceStateChanged() {
 			// Validating if allowed service exists
 			_, err := kube.GetService(service.Address, c.cli.Namespace, c.cli.KubeClient)
 			// If service is now allowed, but does not exist, remove its definition to let service sync recreate it
-			if len(service.Origin) > 0 && err != nil && errors.IsNotFound(err) {
+			if len(service.Origin) > 0 && !service.IsAnnotated() && err != nil && errors.IsNotFound(err) {
 				event.Recordf(c.name, "[validateServiceStateChanged] service is now allowed %s", service.Address)
 				c.cli.ServiceInterfaceRemove(context.Background(), service.Address)
 			}

